skops/cli/_update.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import zipfile
+from pathlib import Path
+
+from skops.cli._utils import get_log_level
+from skops.io import dump, load
+from skops.io._protocol import PROTOCOL
+
+
+def _update_file(
+    input_file: str | Path,
+    output_file: str | Path | None = None,
+    inplace: bool = False,
+    logger: logging.Logger = logging.getLogger(),
+) -> None:
+    """Function that is called by ``skops update`` entrypoint.
+
+    Loads a skops model from the input path, converts to the current skops format, and
+    saves to output file.
+
+    Parameters
+    ----------
+    input_file : str, or Path
+        Path of input skops model to load.
+
+    output_file : str, or Path, default=None
+        Path to save the updated skops model to.
+
+    inplace : bool, default=False
+        Whether to update and overwrite the input file in place.
+
+    logger : logging.Logger, default=logging.getLogger()
+        Logger to use for logging.
+    """
+    if inplace:
+        if output_file is None:
+            output_file = input_file
+        else:
+            raise ValueError(
+                "Cannot specify both an output file path and the inplace flag. Please"
+                " choose whether you want to create a new file or overwrite the input"
+                " file."
+            )
+
+    input_model = load(input_file, trusted=True)
+    with zipfile.ZipFile(input_file, "r") as zip_file:
+        input_file_schema = json.loads(zip_file.read("schema.json"))
+
+    if input_file_schema["protocol"] != PROTOCOL:
+        if output_file is not None:
+            dump(input_model, output_file)
+            logger.info(f"Updated skops file written to {output_file}")
+        else:
+            logger.info(
+                f"File can be updated to the current protocol: {PROTOCOL}. Please"
+                " specify an output file path or use the inplace flag to create the"
+                " updated Skops file."
+            )
+    else:
+        logger.info(
+            "File was not updated because already up to date with the current protocol:"
+            f" {PROTOCOL}"
+        )
+
+
+def format_parser(
+    parser: argparse.ArgumentParser | None = None,
+) -> argparse.ArgumentParser:
+    """Adds arguments and help to parent CLI parser for the `update` method."""
+
+    if not parser:  # used in tests
+        parser = argparse.ArgumentParser()
+
+    parser_subgroup = parser.add_argument_group("update")
+    parser_subgroup.add_argument("input", help="Path to an input file to update.")
+
+    parser_subgroup.add_argument(
+        "-o",
+        "--output-file",
+        help="Specify the output file name for the updated skops file.",
+        default=None,
+    )
+    parser_subgroup.add_argument(
+        "--inplace",
+        help="Update and overwrite the input file in place.",
+        action="store_true",
```

Comment:
Do we want inplace to be the default?

Reply:
I like the idea that `skops update` doesn't do anything (only logs) if someone doesn't specify neither `inplace` nor `output-file`. But I can change it up to you :)

Reply:
That behavior would be fine. It would also be acceptable IMO to update inplace by default, since it would only be done if the current version is higher, but leaving it like this is also good.

Just one question: Right now, if using the defaults (no change to log level), a user would typically not get feedback at all, right? Do we want the default behavior to be not do anything, nor message anything?

Reply:
> Just one question: Right now, if using the defaults (no change to log level), a user would typically not get feedback at all, right? Do we want the default behavior to be not do anything, nor message anything?

Ah yes you are right. Because the messages are logged as "info" and the default logging level should be "warning". They will not show up if one doesn't set `-v`.

Actually we could log certain messages as "warning", not sure if "info" is the correct level for somenthing like [this](https://github.com/skops-dev/skops/pull/343/files#diff-fd06e712870d44acb9dcfbfac570deb5224468b6e9d10277a9c68f4276859d54R68) for example.

Reply:
Yeah, maybe it's better to elevate that message to "warning". As a user, if I call `skops update mymodel.skops` and there is no message whatsoever, I'd assume that the model was successfully updated (if an update was necessary).

Reply:
Cool I changed some levels here: https://github.com/skops-dev/skops/pull/343/commits/b44dca493e64debdff6767fa6bec74617bebb0dd